ethereum/spec/src/test/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/bellatrix/BeaconBlockBodyBellatrixTest.java

```diff
@@ -56,18 +60,56 @@ void equalsReturnsFalseWhenExecutionPayloadIsDifferent() {
     assertNotEquals(defaultBlockBody, testBeaconBlockBody);
   }
 
+  @Test
+  void equalsReturnsFalseWhenExecutionPayloadHeaderIsDifferent() {
+    executionPayloadHeader = dataStructureUtil.randomExecutionPayloadHeader();
+    final BlindedBeaconBlockBodyBellatrix testBlindedBeaconBlockBody =
+        safeJoin(createBlindedBlockBody());
+
+    assertNotEquals(defaultBlindedBlockBody, testBlindedBeaconBlockBody);
+  }
+
+  @Test
+  public void builderShouldFailIfNotPassingPayload() {
```

Comment:
this one should be probably called `builderShouldFailIfNotPassingPayloadAndPayloadHeader`